src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
-
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
 }
 
-export class Triangle {
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue'
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides of a triangle must be pozitive numbers');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Sides '
+      + `${this.a}, ${this.b} and ${this.c} `
```

Comment:
wondering - in conditional block you use a,b,c without `this.`,
but in string literal with `this.` 🙂

Reply:
I didn't realize it , I will give it a look and fix it. 👍
Thank you.